include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -203,6 +208,42 @@ struct erf_helper<FloatingPoint NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScala
 	}
 };
 
+// logn-gamma function
+template<typename T>
+NBL_PARTIAL_REQ_TOP(concepts::FloatingPointScalar<T>)
+struct lgamma_helper<T NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
+{
+	// implementation from Numerical Recipes in C, 2nd ed.
+    static T __call(T val)
+    {
+        T x, y;
+        y = x = val;
+        T tmp = x + T(5.5);
+        tmp -= (x + T(0.5)) * log_helper<T>::__call(tmp);
+
+        T ser = T(1.000000000190015);
+		ser += T(76.18009172947146) / ++y;
+		ser += T(-86.50532032941677) / ++y;
+		ser += T(24.01409824083091) / ++y;
+		ser += T(-1.231739572450155) / ++y;
+		ser += T(0.1208650973866179e-2) / ++y;
+		ser += T(-0.5395239384953e-5) / ++y;
+        return -tmp + log_helper<T>::__call(T(2.5066282746310005) * ser / x);
+    }
+};
+
+// beta function
+template<typename T>
+NBL_PARTIAL_REQ_TOP(concepts::FloatingPointScalar<T>)
+struct beta_helper<T NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
+{
+	// implementation from Numerical Recipes in C, 2nd ed.
+	static T __call(T v1, T v2)
+	{
+		return exp_helper<T>::__call(lgamma_helper<T>::__call(v1) + lgamma_helper<T>::__call(v2) - lgamma_helper<T>::__call(v1+v2));
```

Comment:
implement using `log2` and `exp2`, it requires a modified `l2gamma_helper` which spits out logarithm base 2 return value and not logarithm base e

Reply:
btw in the lgamma I see constants which essentially cancel out, like the 5.5 on the tmp, etc.